l10n_ch_payment_slip/report/ir_action.py

```diff
@@ -1,46 +1,201 @@
+                        'name': save_in_attachment.get(doc.invoice_id.id),
+                        'datas': base64.encodestring(pdfreport.read()),
+                        'datas_fname': save_in_attachment.get(
+                            doc.invoice_id.id),
+                        'res_model': save_in_attachment.get('model'),
+                        'res_id': doc.invoice_id.id,
+                    }
+                    try:
+                        self.env['ir.attachment'].create(attachment)
+                    except AccessError:
+                        _logger.info("Cannot save PDF report %r as attachment",
+                                     attachment['name'])
+                    else:
+                        _logger.info(
+                            'The PDF document %s is now saved in the database',
+                            attachment['name'])
+
+            pdfdocuments.append(pdfreport_path)
+
+        # Return the entire document
+        if len(pdfdocuments) == 1:
+            entire_report_path = pdfdocuments[0]
+        else:
+            if company.merge_mode == 'in_memory':
+                content = self.merge_pdf_in_memory(pdfdocuments)
+                entire_report_path = False
+            else:
+                entire_report_path = self._merge_pdf(pdfdocuments)
+
+        if entire_report_path:
+            with open(entire_report_path, 'rb') as pdfdocument:
+                content = pdfdocument.read()
+            if entire_report_path not in temporary_files:
+                temporary_files.append(entire_report_path)
+
+        # Manual cleanup of the temporary files
+        for temporary_file in temporary_files:
+            try:
+                os.unlink(temporary_file)
+            except (OSError, IOError):
+                _logger.error(
+                    'Error when trying to remove file %s' % temporary_file)
+
+        return content
+
     @api.model
-    def _lookup_report(self, name):
-        self.env.cr.execute(
-            "SELECT * FROM ir_act_report_xml WHERE report_name=%s", (name,))
-        report = self.env.cr.dictfetchone()
-        if report and report['report_type'] == 'reportlab-pdf':
-            return report['report_name']
+    def _get_report_from_name(self, report_name):
+        """Return also report of report_type reportlab-pdf and not only qweb
+        """
+        report = super()._get_report_from_name(report_name)
+        if not report:
+            report = report.search([
+                ('report_type', '=', 'reportlab-pdf'),
+                ('report_name', '=', report_name)],
+                limit=1
+            )
+        return report
+
+    @api.multi
+    def render_qweb_pdf(self, res_ids=None, data=None):
+        if (self.name == 'l10n_ch_payment_slip.'
+                         'one_slip_per_page_from_invoice'):
+            report = self._get_report_from_name(self.name)
+            save_in_attachment = self._check_attachment_use(
```

Comment:
_check_attachment_use does not exist anymore in Odoo 11.0.

This was removed here 

https://github.com/odoo/odoo/commit/e80238042c9d93d492ea8b06b0041aced0d81dcd#diff-c941fd9fa97231fc06b29633532dd307L318